// @aws-sdk/client-sagemaker/dist-es/models/models_5.js
import { SENSITIVE_STRING } from "@smithy/smithy-client";
import { ModelPackageModelCardFilterSensitiveLog, OidcConfigFilterSensitiveLog, } from "./models_2";
export const ListWorkteamsSortByOptions = {
    CreateDate: "CreateDate",
    Name: "Name",
};
export const ModelVariantAction = {
    PROMOTE: "Promote",
    REMOVE: "Remove",
    RETAIN: "Retain",
};
export const Relation = {
    EQUAL_TO: "EqualTo",
    GREATER_THAN_OR_EQUAL_TO: "GreaterThanOrEqualTo",
};
export const SearchSortOrder = {
    ASCENDING: "Ascending",
    DESCENDING: "Descending",
};
export const VariantPropertyType = {
    DataCaptureConfig: "DataCaptureConfig",
    DesiredInstanceCount: "DesiredInstanceCount",
    DesiredWeight: "DesiredWeight",
};
export const ModelCardFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Content && { Content: SENSITIVE_STRING }),
});
export const ModelPackageFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.ModelCard && { ModelCard: ModelPackageModelCardFilterSensitiveLog(obj.ModelCard) }),
});
export const SearchRecordFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.TrialComponent && { TrialComponent: obj.TrialComponent }),
    ...(obj.ModelPackage && { ModelPackage: ModelPackageFilterSensitiveLog(obj.ModelPackage) }),
    ...(obj.FeatureGroup && { FeatureGroup: obj.FeatureGroup }),
    ...(obj.ModelCard && { ModelCard: ModelCardFilterSensitiveLog(obj.ModelCard) }),
});
export const SearchResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Results && { Results: obj.Results.map((item) => SearchRecordFilterSensitiveLog(item)) }),
});
export const UpdateModelCardRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Content && { Content: SENSITIVE_STRING }),
});
export const UpdateModelPackageInputFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.ModelCard && { ModelCard: ModelPackageModelCardFilterSensitiveLog(obj.ModelCard) }),
});
export const UpdateWorkforceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.OidcConfig && { OidcConfig: OidcConfigFilterSensitiveLog(obj.OidcConfig) }),
});
